src/main/java/com/odysseusinc/arachne/portal/config/tenancy/TenantContext.java

```diff
@@ -1,18 +1,24 @@
 package com.odysseusinc.arachne.portal.config.tenancy;
 
-import org.apache.commons.lang3.ObjectUtils;
+import com.odysseusinc.arachne.portal.model.security.ArachneUser;
+import java.util.Optional;
+import org.springframework.security.core.context.SecurityContextHolder;
 
 public class TenantContext {
 
-    private static ThreadLocal<Long> currentTenant = new ThreadLocal<>();
-
     public static Long getCurrentTenant() {
-
-        return currentTenant.get();
+        return Optional.ofNullable(
+                SecurityContextHolder.getContext().getAuthentication()
+        ).map(authentication -> {
+            Object principal = authentication.getPrincipal();
+            return  (principal instanceof ArachneUser) ? ((ArachneUser) principal).getActiveTenantId() : null;
+        }).orElse(null);
     }
 
+    /**
+     * @deprecated Does nothing. Tenant is now provided from user information stored in SecurityContext.
+     * Used only in tests and should be removed there.
+     */
```

Comment:
User should be able to switch between tenants when belongs to more than one.
How that could be done?

Reply:
Switching tenants has nothing to do with this method. When it occurs, `active_tenant_id` is set in `user_data`.
This method was merely responsible only for storing active tenant id from DB into a thread local value as part of filter chain.
This is excessive, as ArachneUser is now already fully stored  in SecurityContext  (which is the better way to do it compared to ThreadLocal as Spring takes care to ensure it is cleared and no cross-request contamination should be possible) so the TenantContext now provides it from there.

Reply:
Got it. Thank you